modules/vetext/app/models/vetext/v0/vaccine_registry.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require 'active_model'
+require 'common/models/form'
+require 'common/exceptions'
+
+module Vetext
+  module V0
+    class VaccineRegistry < Common::Form
+      UNASSIGNABLE_ATTRIBUTES = %i[patient_icn sta3n sta6a authenticated].freeze
+
+      attribute :patient_icn, String            # Not Assignable by params
+      attribute :sta3n, String                  # Not Assignable by params
+      attribute :sta6a, String                  # Not Assignable by params
+      attribute :authenticated, Boolean         # Not Assignable by params
+      attribute :vaccine_interest, String
+      attribute :date_vaccine_received, String
+      attribute :contact, Boolean
+      attribute :contact_method, String
+      attribute :reason_undecided, String
+      attribute :phone, String
+      attribute :email, String
+      attribute :first_name, String             # Overridden by MVI
+      attribute :middle_initial, String         # TODO: VEText MISSING ATTRIBUTE (needed for MVI lookup)
+      attribute :last_name, String              # Overridden by MVI
+      attribute :gender, String                 # TODO: VEText MISSING ATTRIBUTE (needed for MVI lookup)
+      attribute :date_of_birth, String          # Overridden by MVI
+      attribute :patient_ssn, String            # Overridden by MVI
+
+      attr_reader :user
+
+      def initialize(attributes, user = nil)
+        super(attributes.except(*UNASSIGNABLE_ATTRIBUTES))
+        @user = user || unauthenticated_user
+      end
+
+      def unauthenticated_user
+        # TODO: figure out how to fetch from MVI when unauthenticated_user
```

Comment:
you could do someting like this https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/models/health_care_application.rb#L121

Reply:
or, do you have EDIPI or ICN? 

this would do (though MPI might not be cool with it): 
MPI::Service.new.find_profile(OpenStruct.new(mhv_icn: the_icn, dslogon_edipi: the_edipi)).profile

Reply:
We would only have ICN if the user has already signed in, in which case we wouldn't need to do the lookup because we have everything we need already from SSOe or MVI. However we want to try to do an MVI lookup to provide ICN and other details when the user is NOT logged in as well, but we have to do it in a way that doesn't return success/failure, possibly even use a CAPTCHA or something to avoid abuse. The idea would be that we do the MVI lookup and send the appropriate attributes over based on a successful lookup of the attributes the user provides as part of the form... it would be an attribute based lookup in MVI.